### PR TITLE
feat: Add guardrails span kind

### PR DESCRIPTION
**Pull request by nate-mar**
Add guardrails span kind to list of span kinds

### CODE REVIEW

**Comment by nate-mar** [comment]
I have read the CLA Document and I hereby sign the CLA

File: js/packages/openinference-semantic-conventions/src/trace/SemanticConventions.ts [hunk 1]
```diff
@@ -341,6 +341,7 @@ export enum OpenInferenceSpanKind {
   RERANKER = "reranker",
   EMBEDDING = "embedding",
   AGENT = "agent",
+  GUARDRAILS = "guardrails",
```
Comment (mikeldking) [inline js/packages/openinference-semantic-conventions/src/trace/SemanticConventions.ts:344]:
```suggestion
  GUARDRAIL = "guardrail",
```
Can we make it singular? Also what would be the typical contents of a guardrail span?

**Review by mikeldking — CHANGES_REQUESTED** [review]
Let's add spec changes and also python changes

**Comment by nate-mar** [comment]
@mikeldking will do -- can you point me to where in the spec file I should add this span kind? I can't seem to locate a place in the spec where these span kinds are comprehensively listed; I only see a list of example span kinds under the `openinference.span_kind` attribute.

**Comment by mikeldking** [comment]
> @mikeldking will do -- can you point me to where in the spec file I should add this span kind? I can't seem to locate a place in the spec where these span kinds are comprehensively listed; I only see a list of example span kinds under the `openinference.span_kind` attribute.

Yes, as this is technically an attribute value it's not bound by any convention. But at least describing the value here would be ideal: https://github.com/Arize-ai/openinference/blob/main/spec/traces.md

File: js/packages/openinference-semantic-conventions/src/trace/SemanticConventions.ts [hunk 2]
```diff
@@ -341,6 +341,7 @@ export enum OpenInferenceSpanKind {
   RERANKER = "reranker",
   EMBEDDING = "embedding",
   AGENT = "agent",
+  GUARDRAIL = "guardrail",
```
Comment (mikeldking) [inline js/packages/openinference-semantic-conventions/src/trace/SemanticConventions.ts:344]: These changes require something called a changeset to be deployed: https://github.com/Arize-ai/openinference/blob/main/js/DEVELOPMENT.md#changesets

**Review by mikeldking — APPROVED** [review]
Exciting!

**Pull request merged**